args.go

```diff
@@ -99,3 +75,13 @@ func RangeArgs(min int, max int) PositionalArgs {
 		return nil
 	}
 }
+
+// ExactValidArgs is DEPRECATED. Use ExactArgs instead.
```

Comment:
I'm sorry, I should have looked more closely the first time. So these aren't just deprecated, you literally broke them? While my code would still compile it wouldn't work correctly anymore. Let's say my code uses the arg to look up a value in a slice. After this patch my code would start to panic() on invalid input instead of failing nicely.

That's not a deprecation, that's API breakage and should only be done until cobra v2. Deprecation means you keep it working, but you tell people you are going to break it eventually.

OnlyValidArgs and ExactValidArgs are exported functions which are part of our API. They must remain working and must remain tested.

Please add a comment line that starts with `// Deprecated: ` similarly to what the core library uses to indicate deprecation. While there aren't many tools that look for it at least a couple do and it makes things a little more consistent for users.

Reply:
> I'm sorry, I should have looked more closely the first time. So these aren't just deprecated, you literally broke them? While my code would still compile it wouldn't work correctly anymore.

That was definitely not expected. I will have a look.

At first I wanted to remove them, not just deprecate. So, yes, this PR was meant to be released in v2. However, since @jharshman suggested it, I then tried to keep then functional.

> Let's say my code uses the arg to look up a value in a slice. After this patch my code would start to panic() on invalid input instead of failing nicely.

I assume that this use case is not covered by any of the existing tests. Precisely, I now added `TestOnlyValid*` and `TestExactValid*` tests again, and all of them are successful. Is it possible to have a MWE to work on?
 
> That's not a deprecation, that's API breakage and should only be done until cobra v2. Deprecation means you keep it working, but you tell people you are going to break it eventually.

As said, I'm ok with any of them. Nonetheless, in this case I think that deprecation without breakage is possible.

> OnlyValidArgs and ExactValidArgs are exported functions which are part of our API. They must remain working and must remain tested.

As commented above, I am keeeping the tests now, and all of them are successful.

> Please add a comment line that starts with `// Deprecated: ` similarly to what the core library uses to indicate deprecation. While there aren't many tools that look for it at least a couple do and it makes things a little more consistent for users.

I didn't find any example in the repo (`grep -r ./ -e '// Deprecated: '`). Do you mean [golang/go/wiki/Deprecated](https://github.com/golang/go/wiki/Deprecated)?

Reply:
> At first I wanted to remove them, not just deprecate. So, yes, this PR was meant to be released in v2. However, since @jharshman suggested it, I then tried to keep then functional.

Yes is it important to keep them functional.

Reply:
> Yes is it important to keep them functional.

Could you reproduce the `panic` that @eparis commented?